### PR TITLE
ta: avb: copy data to temporary buffers

### PR DESCRIPTION
**Pull request by igoropaniuk**
```
Use intermediate temporary buffers instead of directly supplying non-secure
buffers to TEE_ReadObjectData()/TEE_CreatePersistentObject().
This fixes TA panics while accessing persistent objects:

E/TC:? 0 TA panicked with code 0xffff0001
E/LD:  Status of TA 023f8f1a-292a-432b-8fc4-de8471358067
...
D/TC:? 0 user_ta_enter:176 tee_user_ta_enter: TA panicked with
code 0xffff0001

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
```

### CODE REVIEW

File: ta/avb/entry.c [hunk 1]
```diff
@@ -267,8 +267,14 @@ static TEE_Result write_persist_value(uint32_t pt,
...
+	char *value = TEE_Malloc(value_sz, 0);
+
+	if (!value)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	TEE_MemMove(value, params[1].memref.buffer,
+		    value_sz);
 
 	res = get_named_object_name(name_buf, name_buf_sz,
 				    name_full, &name_full_sz);
```
Comment (jenswi-linaro) [inline ta/avb/entry.c:280]: Memory leakage in the error handling below.

File: ta/avb/entry.c [hunk 2]
```diff
@@ -309,8 +316,11 @@ static TEE_Result read_persist_value(uint32_t pt,
...
 
-	char *value = params[1].memref.buffer;
 	uint32_t value_sz = params[1].memref.size;
+	char *value = TEE_Malloc(value_sz, 0);
+
+	if (!value)
+		return TEE_ERROR_OUT_OF_MEMORY;
 
 	res = get_named_object_name(name_buf, name_buf_sz,
 				    name_full, &name_full_sz);
```
Comment (jenswi-linaro) [inline ta/avb/entry.c:326]: Memory leakage in the error handling of TEE_OpenPersistentObject() below.

**Comment by igoropaniuk** [comment]
pushed updated version

File: ta/avb/entry.c [hunk 3]
```diff
@@ -330,9 +340,14 @@ static TEE_Result read_persist_value(uint32_t pt,
 		goto out;
 	}
 
+	TEE_MemMove(params[1].memref.buffer, value,
+		    value_sz);
+
 	params[1].memref.size = count;
 out:
 	TEE_CloseObject(h);
+memfree:
```
Comment (jenswi-linaro) [inline ta/avb/entry.c:349]: We'd normally call this label `out_free:`.

File: ta/avb/entry.c [hunk 4]
```diff
@@ -283,6 +288,8 @@ static TEE_Result write_persist_value(uint32_t pt,
 		EMSG("Can't create named object value, res = 0x%x", res);
 
 	TEE_CloseObject(h);
+memfree:
```
Comment (jenswi-linaro) [inline ta/avb/entry.c:291]: We'd normally call this label `out:` or `out_free:`.

**Review by jenswi-linaro — COMMENTED** [review]
With or without my two minor remarks addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by igoropaniuk** [comment]
renamed labes, tag applied

**Pull request merged**